### PR TITLE
note that views works on strings too

### PR DESCRIPTION
**Pull request by stevengj**
This seemed to be an omission in the manual.

### CODE REVIEW

File: doc/src/manual/strings.md [hunk 1]
```diff
@@ -254,6 +255,9 @@ julia> substr = SubString(str, 1, 4)
 
 julia> typeof(substr)
 SubString{String}
+
+julia> @views typeof(str[1:4]) # @views converts slices to SubStrings
```
Comment (simeonschaub) [inline doc/src/manual/strings.md:259]: Might be a bit clearer to put the `@views` inside the `typeof`?

**Review by simeonschaub — APPROVED** [review]
LGTM

File: doc/src/manual/strings.md (hunk 1 above)
Comment (stevengj) [reply doc/src/manual/strings.md:259]: The **whole point** of `@views` is that you *don't* need to put it directly adjacent to the indexing expression (unlike `@view`).

I think many people aren't aware of this feature of `@views`, and so it is important to use it differently from `@view` in examples.

File: base/strings/substring.jl [hunk 2]
```diff
@@ -7,6 +7,9 @@
 Like [`getindex`](@ref), but returns a view into the parent string `s`
 within range `i:j` or `r` respectively instead of making a copy.
 
+The [`@views`](@ref) macro converts any string slices `s[i:j]` into
```
Comment (fredrikekre) [inline base/strings/substring.jl:10]: Since the same is true for `@view` too, perhaps mention that too?

**Pull request merged**